src/Agent.Worker/TaskManager.cs

```diff
@@ -109,6 +116,24 @@ public Definition Load(Pipelines.TaskStep task)
             return definition;
         }
 
+        public void Extract(IExecutionContext executionContext, Pipelines.TaskStep task)
+        {
+            String zipFile = GetTaskZipPath(task.Reference);
+            String destinationDirectory = GetDirectory(task.Reference);
+            
+            executionContext.Debug($"Extracting task {task.Name} from {zipFile} to {destinationDirectory}.");
+            
+            IOUtil.DeleteDirectory(destinationDirectory, CancellationToken.None);
```

Comment:
are we sure we don't have a cancellation token we can pass into these operations?

Reply:
There is an `IHostContext.AgentShutdownToken` and `ExecutionContext.CancellationToken`. Any idea which would be better? I am thinking the latter based on significant usage in code.

I updated this to use the latter but let me know if you think that's wrong.

Reply:
I am going to pass it to VerifyAsync and pass that to the process invoker for NuGet too.